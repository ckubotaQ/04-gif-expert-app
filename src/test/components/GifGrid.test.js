import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas en <GifGrid />', () => {
    const category= "digimon";
   
    test('Debe de coincidir con el SS', () => {
        useFetchGifs.mockReturnValue({ data: [], loading: true
        })
        const wrapper = shallow(<GifGrid category={category}/>)
             expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar items cuando se carga imagen useFecthGifd', () => {
        const imgs = [{
            id: 'abc',
            title:'Digimon',
            url: "://https:localhost/casilo"
        },{
            id: 'abqwwc',
            title:'Digimon',
            url: "://https:localhost/casilo"
        }]
        useFetchGifs.mockReturnValue({ data: imgs, loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);

    })
    



})
