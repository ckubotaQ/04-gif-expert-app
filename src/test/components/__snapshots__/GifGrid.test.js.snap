// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de coincidir con el SS 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    digimon
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
